src/gridfs/options.rs

```diff
@@ -0,0 +1,92 @@
+#[non_exhaustive]
+pub struct GridFsBucketOptions {
+    /// The bucket name. Defaults to 'fs'.
+    pub bucket_name: Option<String>,
+
+    /// The chunk size in bytes used to break the user file into chunks. Defaults to 255 KiB.
+    pub chunk_size_bytes: Option<i32>,
+
+    /// The write concern. Defaults to the write concern of the database.
+    pub write_concern: Option<WriteConcern>,
+
+    /// The read concern. Defaults to the read concern of the database.
+    pub read_concern: Option<ReadConcern>,
+
+    /// The read preference. Defaults to the read preference of the database.
+    pub read_preference: Option<SelectionCriteria>,
+}
+
+/// Contains the options for creating a [`GridFsStream`] to upload a file to a
+/// [`GridFsBucket`].
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[builder(field_defaults(setter(into)))]
+#[non_exhaustive]
+pub struct GridFsUploadOptions {
+    /// The number of bytes per chunk of this file. Defaults to the `chunk_size_bytes` specified
+    /// in the [`GridFsBucketOptions`].
+    pub chunk_size_bytes: Option<i32>,
+
+    /// User data for the 'metadata' field of the files collection document.
+    pub metadata: Option<Document>,
+}
+
+/// Contains the options for creating [`GridFsStream`] to retrieve a stored file
+/// from a [`GridFsBucket`].
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[builder(field_defaults(setter(into)))]
+#[non_exhaustive]
+pub struct GridFsDownloadByNameOptions {
+    /// Which revision (documents with the same filename and different `upload_date`)
+    /// of the file to retrieve. Defaults to -1 (the most recent revision).
+    ///
+    /// Revision numbers are defined as follows:
+    /// 0 = the original stored file
+    /// 1 = the first revision
+    /// 2 = the second revision
+    /// etc...
+    /// -2 = the second most recent revision
+    /// -1 = the most recent revision
+    pub revision: Option<i32>,
+}
+
+/// Contains the options for performing a find operation on a files collection.  
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[builder(field_defaults(setter(into)))]
+#[non_exhaustive]
+pub struct GridFsFindOptions {
+    /// Enables writing to temporary files on the server. When set to true, the
+    /// server can write temporary data to disk while executing the find operation
+    /// on the files collection.
+    pub allow_disk_use: Option<bool>,
+
+    /// The number of documents to return per batch.
+    pub batch_size: Option<i32>,
+
+    /// The maximum number of documents to return.
+    pub limit: Option<i32>,
+
+    /// The maximum amount of time to allow the query to run.
+    pub max_time_ms: Option<i64>,
+
+    /// The server normally times out idle cursors after an inactivity period
+    /// (10 minutes) to prevent excess memory use. Set this option to prevent that.
+    pub no_cursor_timeout: Option<bool>,
+
+    /// The number of documents to skip before returning.
+    pub skip: i32,
```

Comment:
I know the spec doesn't say it but I think this should probably be optional? also, this is a `u64` for `FindOptions` so I think it would make sense to do the same here